crates/trident/src/engine/update.rs

```diff
@@ -145,8 +145,29 @@ pub(crate) fn update(
                 );
                 Ok(ExitKind::Done)
             } else {
-                runtime_update::finalize_update(&mut subsystems, state, Some(update_start_time))
-                    .message("Failed to finalize runtime update")
+                let finalize_result = runtime_update::finalize_update(
+                    &mut subsystems,
+                    state,
+                    false,
+                    true,
+                    Some(update_start_time),
+                )
+                .message("Failed to finalize runtime update");
+                if let Err(e) = finalize_result {
+                    error!("Runtime update finalize failed with message: {e:?}");
+                    // Attempt an auto-rollback
```

Comment:
expand this comment a little to describe how calling finalize_update a second time is attempting an auto-rollback